src/components/AnnotationCreator.jsx

```diff
@@ -19,27 +19,91 @@ import usePostAnnotation from '../models/annotation/usePostAnnotation';
 import Button from './Button';
 import Text from './Text';
 import StandardDialog from './StandardDialog';
-import CustomAlert from './Alert';
+import Alert from './Alert';
+import useAddEncounter from '../models/encounter/useAddEncounter';
+import useAddAnnotationsToSightingEncounter from '../models/encounter/useAddAnnotationsToSightingEncounter';
 
 function percentageToPixels(percentValue, scalar) {
   const pixelValue = 0.01 * scalar * percentValue;
   return Math.round(Math.max(pixelValue, 0));
 }
 
+async function createEncounter(sightingData, addEncounterToSighting) {
+  const copiedProperties = pick(sightingData, [
+    'time',
+    'timeSpecificity',
+    'locationId',
+  ]);
+  const sightingId = get(sightingData, 'guid');
+  const encounterGuidsBeforeCreation = map(
+    get(sightingData, 'encounters'),
+    encounter => encounter?.guid,
+    [],
+  );
+  const encounterCreationResponse = await addEncounterToSighting({
+    sightingGuid: sightingId,
+    operations: [
+      {
+        op: 'add',
+        path: '/encounters',
+        value: copiedProperties,
+      },
+    ],
+  });
+  const encounterGuidsAfterCreation = map(
+    get(encounterCreationResponse, ['data', 'encounters'], []),
+    encounter => encounter?.guid,
+    [],
+  );
+  const newEncounterGuids = filter(
+    encounterGuidsAfterCreation,
+    enc => encounterGuidsBeforeCreation.indexOf(enc) === -1,
+    [],
+  );
+  return encounterCreationResponse?.status === 200 &&
+    newEncounterGuids?.length === 1
+    ? newEncounterGuids
+    : [];
+}
+
 export default function AnnotationCreator({
   titleId = 'CREATE_ANNOTATION',
   asset,
   onClose,
   sightingData,
-  refreshSightingData,
+  pending,
 }) {
   const [viewpoint, setViewpoint] = useState('');
   const [IAClass, setIAClass] = useState('');
   const [rect, setRect] = useState({});
   const [anchorEl, setAnchorEl] = useState(null);
   const theme = useTheme();
-
-  const { postAnnotation, loading, error } = usePostAnnotation();
+  const {
+    mutate: addEncounterToSighting,
+    isLoading: addEncounterToSightingLoading,
+    error: addEncounterToSightingError,
+  } = useAddEncounter();
+  const {
+    addAnnotationsToSightingEncounter,
+    error: addAnnotationsToSightingEncounterError,
+    isLoading: addAnnotationToSightingEncounterLoading,
+  } = useAddAnnotationsToSightingEncounter();
+  const {
+    mutate: postAnnotation,
+    isLoading: postAnnotationLoading,
+    error: postAnnotationError,
+  } = usePostAnnotation();
+  const loading =
+    addEncounterToSightingLoading ||
+    addAnnotationToSightingEncounterLoading ||
+    postAnnotationLoading;
+  const hasErrors =
+    addEncounterToSightingError ||
+    addAnnotationsToSightingEncounterError ||
+    postAnnotationError;
+  const onCancel = () => {
+    onClose();
+  };
```

Comment:
Do the mutation errors need to be cleared here?

Reply:
Apparently not! They were gone when I either cancelled or dismissed the dialog. I was similarly perplexed!